src/styles/main.scss

```diff
@@ -1,3 +1,71 @@
+$day-size: 100px;
+$space-between-days: 1px;
+$calendar-padding: 10px;
+$calendar-width:
+  7 * ($day-size + $space-between-days) - 1 + 2 * $calendar-padding;
+$days: ("mon": 1, "tue": 2, "wed": 3, "thu": 4, "fri": 5, "sat": 6, "sun": 7);
+
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
 body {
   margin: 0;
 }
+
+.calendar {
+  width: $calendar-width;
+  padding: $calendar-padding;
+  display: flex;
+  align-items: center;
+  flex-wrap: wrap;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  gap: $space-between-days;
+
+  &--start-day {
+    @each $day, $day-num in $days {
+      &_#{$day} :first-child {
+        margin-left: ($day-num - 1) * ($day-size + $space-between-days);
+      }
+    }
+  }
+
+  &--month-length {
+    @for $length from 28 to 32 {
+      &_#{$length} :nth-child(n + #{$length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @for $d from 1 to 32 {
```

Comment:
please change this variable name to be more descriptible `@for $day from 1 to 32 {`